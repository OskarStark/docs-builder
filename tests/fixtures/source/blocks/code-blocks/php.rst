.. code-block:: php

    // config/routes.php
    namespace Symfony\Component\Routing\Loader\Configurator;

    use App\Controller\CompanyController;

    return static function (RoutingConfigurator $routes): void {
        $routes->add('about_us', ['nl' => '/over-ons', 'en' => '/about-us'])
            ->controller(CompanyController::class.'::about');
    };

.. code-block:: php

    enum TextAlign: string implements TranslatableInterface
    {
        case Left = 'Left aligned';
        case Center = 'Center aligned';
        case Right = 'Right aligned';

        public function trans(TranslatorInterface $translator, ?string $locale = null): string
        {
            // Translate enum using custom labels
            return match ($this) {
                self::Left => $translator->trans('text_align.left.label', locale: $locale),
                self::Center => $translator->trans('text_align.center.label', locale: $locale),
                self::Right => $translator->trans('text_align.right.label', locale: $locale),
            };
        }
    }

.. code-block:: php

    public function getUserBadgeFrom(string $accessToken): UserBadge
    {
        // get the data from the token
        $payload = ...;

        return new UserBadge(
            $payload->getUserId(),
            fn (string $userIdentifier): User => new User($userIdentifier, $payload->getRoles())
        );

        // or
        return new UserBadge(
            $payload->getUserId(),
            $this->loadUser(...)
        );
    }
