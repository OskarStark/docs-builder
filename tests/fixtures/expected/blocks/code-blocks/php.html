<div translate="no" data-loc="9" class="notranslate codeblock codeblock-length-sm codeblock-php">
    <div class="codeblock-scroll">
        <pre class="codeblock-lines">1
2
3
4
5
6
7
8
9</pre>
        <pre class="codeblock-code"><code><span class="hljs-comment">// config/routes.php</span>
<span class="hljs-keyword">namespace</span> <span class="hljs-title">Symfony</span>\<span class="hljs-title">Component</span>\<span class="hljs-title">Routing</span>\<span class="hljs-title">Loader</span>\<span class="hljs-title">Configurator</span>;

    <span class="hljs-keyword">use</span> <span class="hljs-title">App</span>\<span class="hljs-title">Controller</span>\<span class="hljs-title">CompanyController</span>;

<span class="hljs-keyword">return</span> <span class="hljs-keyword">static</span> <span class="hljs-function"><span class="hljs-keyword">function</span> <span class="hljs-params">(RoutingConfigurator <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>routes</span>)</span>: <span class="hljs-title">void</span> </span>{
    <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>routes</span>-&gt;<span class="hljs-title invoke__">add</span>(<span class="hljs-string">'about_us'</span>, [<span class="hljs-string">'nl'</span> =&gt; <span class="hljs-string">'/over-ons'</span>, <span class="hljs-string">'en'</span> =&gt; <span class="hljs-string">'/about-us'</span>])
        -&gt;<span class="hljs-title invoke__">controller</span>(CompanyController::<span class="hljs-variable language_">class</span>.<span class="hljs-string">'::about'</span>);
};</code></pre>
    </div>
</div>
<div translate="no" data-loc="16" class="notranslate codeblock codeblock-length-md codeblock-php">
    <div class="codeblock-scroll">
        <pre class="codeblock-lines">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16</pre>
        <pre class="codeblock-code"><code>
<span class="hljs-class"><span class="hljs-keyword">enum</span> <span class="hljs-title">TextAlign</span>: <span class="hljs-title">string</span> <span class="hljs-keyword">implements</span> <span class="hljs-title">TranslatableInterface</span></span>
{
    <span class="hljs-keyword">case</span> Left = <span class="hljs-string">'Left aligned'</span>;
    <span class="hljs-keyword">case</span> Center = <span class="hljs-string">'Center aligned'</span>;
    <span class="hljs-keyword">case</span> Right = <span class="hljs-string">'Right aligned'</span>;

    <span class="hljs-keyword">public</span> <span class="hljs-function"><span class="hljs-keyword">function</span> <span class="hljs-title">trans</span><span class="hljs-params">(TranslatorInterface <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>translator</span>, ?<span class="hljs-keyword">string</span> <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>locale</span> = <span class="hljs-keyword">null</span>)</span>: <span class="hljs-title">string</span></span>
    {
        <span class="hljs-comment">// Translate enum using custom labels</span>
        <span class="hljs-keyword">return</span> <span class="hljs-keyword">match</span> (<span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>this</span>) {
            <span class="hljs-keyword">self</span>::<span class="hljs-variable constant_">Left</span> =&gt; <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>translator</span>-&gt;<span class="hljs-title invoke__">trans</span>(<span class="hljs-string">'text_align.left.label'</span>, <span class="hljs-attr">locale</span>: <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>locale</span>),
            <span class="hljs-keyword">self</span>::<span class="hljs-variable constant_">Center</span> =&gt; <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>translator</span>-&gt;<span class="hljs-title invoke__">trans</span>(<span class="hljs-string">'text_align.center.label'</span>, <span class="hljs-attr">locale</span>: <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>locale</span>),
            <span class="hljs-keyword">self</span>::<span class="hljs-variable constant_">Right</span> =&gt; <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>translator</span>-&gt;<span class="hljs-title invoke__">trans</span>(<span class="hljs-string">'text_align.right.label'</span>, <span class="hljs-attr">locale</span>: <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>locale</span>),
        };
    }
}
            </code>
        </pre>
    </div>
</div>
<div translate="no" data-loc="16" class="notranslate codeblock codeblock-length-md codeblock-php">
    <div class="codeblock-scroll">
        <pre class="codeblock-lines">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16</pre>
        <pre class="codeblock-code"><code>
<span class="hljs-keyword">public</span> <span class="hljs-function"><span class="hljs-keyword">function</span> <span class="hljs-title">getUserBadgeFrom</span><span class="hljs-params">(<span class="hljs-keyword">string</span> <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>accessToken</span>)</span>: <span class="hljs-title">UserBadge</span></span>
{
    <span class="hljs-comment">// get the data from the token</span>
    <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>payload</span> = ...;

    <span class="hljs-keyword">return</span> <span class="hljs-keyword">new</span> <span class="hljs-title invoke__">UserBadge</span>(
        <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>payload</span>-&gt;<span class="hljs-title invoke__">getUserId</span>(),
        <span class="hljs-function"><span class="hljs-keyword">fn</span> <span class="hljs-params">(<span class="hljs-keyword">string</span> <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>userIdentifier</span>)</span>: <span class="hljs-title">User</span> =&gt;</span> <span class="hljs-keyword">new</span> <span class="hljs-title invoke__">User</span>(<span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>userIdentifier</span>, <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>payload</span>-&gt;<span class="hljs-title invoke__">getRoles</span>())
    );

    <span class="hljs-comment">// or</span>
    <span class="hljs-keyword">return</span> <span class="hljs-keyword">new</span> <span class="hljs-title invoke__">UserBadge</span>(
        <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>payload</span>-&gt;<span class="hljs-title invoke__">getUserId</span>(),
        <span class="hljs-variable"><span class="hljs-variable-other-marker">$</span>this</span>-&gt;<span class="hljs-title invoke__">loadUser</span>(...)
    );
}
        </code></pre>
    </div>
</div>
